File: 053/main.cpp
#include <iostream>
#include <vector>
#include <math.h>
using namespace std;

class Solution
{
  public:
    int maxSubArray(vector<int> &nums)
    {
        int maxToCur = nums[0];
        int res = nums[0];
        for (int i = 1; i < nums.size(); i++)
        {
            maxToCur = max(maxToCur + nums[i], nums[i]);
            res = max(res, maxToCur);
        }
        return res;
    }
};

int main()
{
    Solution sol;
    int res;
    vector<int> nums;
    nums.push_back(-2);
    nums.push_back(1);
    nums.push_back(-3);
    nums.push_back(4);
    nums.push_back(-1);
    nums.push_back(2);
    nums.push_back(1);
    nums.push_back(-5);
    nums.push_back(4);
    res = sol.maxSubArray(nums);
    cout << res << endl;
    return 0;
}

File: 053/main.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {
  const len = nums.length
  let maxToCur = nums[0]
  let max = nums[0]
  for (let i = 1; i < len; i++) {
    maxToCur = Math.max(maxToCur + nums[i], nums[i])
    max = Math.max(max, maxToCur)
  }
  return max
}

console.log(maxSubArray([-2, 1, -3, 4, -1, 2, 1, -5, 4]))
